app/components/post_draft/draft_input/index.tsx

```diff
@@ -133,16 +133,102 @@ export default function DraftInput({
     const handleLayout = useCallback((e: LayoutChangeEvent) => {
         updatePostInputTop(e.nativeEvent.layout.height);
     }, []);
 
+    // To keep track of cursor position and to retrieve the latest value of cursor position
+    const cursorPositionRef = useRef(cursorPosition);
+    useEffect(() => {
+        cursorPositionRef.current = cursorPosition;
+    }, [cursorPosition]);
+
+    const handleEmojiPress = (emojiName: string) => {
+        updateValue((v) => {
+            const name = emojiName.trim();
+            const currentCursorPosition = cursorPositionRef.current;
+            let unicode;
+            const imageUrl = '';
+            if (EmojiIndicesByAlias.get(name)) {
+                const emoji = Emojis[EmojiIndicesByAlias.get(name)!];
+                if (emoji.category === 'custom') {
+                    return `${v.slice(0, currentCursorPosition)} :${emojiName}: ${v.slice(currentCursorPosition)}`;
+                }
+                unicode = emoji.image;
+                if (unicode && !imageUrl) {
+                    const codeArray = unicode.split('-');
+                    // eslint-disable-next-line max-nested-callbacks
+                    const code = codeArray.reduce((acc: string, c: string) => {
+                        return acc + String.fromCodePoint(parseInt(c, 16));
+                    }, '');
+                    return v.slice(0, currentCursorPosition) + code + v.slice(currentCursorPosition);
+                }
+            }
+            return `${v.slice(0, currentCursorPosition)} :${emojiName}: ${v.slice(currentCursorPosition)}`;
+        });
+    };
+
     const inputRef = useRef<PasteInputRef>();
+
     const focus = useCallback(() => {
         inputRef.current?.focus();
     }, []);
 
+    const handleToggleEmojiPicker = () => {
+        if (!isEmojiPickerOpen) {
+            Keyboard.dismiss();
+            setIsEmojiPickerOpen(true);
+            setIsEmojiPickerFocused(true);
+            inputRef.current?.setNativeProps({
+                showSoftInputOnFocus: false,
+            });
+
+            // To make sure the state is updated before focusing
+            setTimeout(() => {
+                focus();
+            }, 0);
+            return;
+        }
+        if (Platform.OS === 'android' && isEmojiPickerFocused) {
+            setIsEmojiPickerFocused(false);
+            setIsEmojiPickerOpen(false);
+            inputRef.current?.setNativeProps({
+                showSoftInputOnFocus: true,
+            });
+
+            inputRef.current?.blur();
+
+            // To make sure the state is updated before focusing
+            setTimeout(() => {
+                focus();
+            }, 0);
+            return;
+        }
+        setIsEmojiPickerFocused(false);
+        setIsEmojiPickerOpen(false);
+        inputRef.current?.setNativeProps({
+            showSoftInputOnFocus: true,
+        });
+        focus();
+    };
+
+    const deleteCharFromCurrentCursorPosition = () => {
```

Comment:
This is passed as a prop, so please use `useCallback`.

Reply:
Done.